skills_ml/evaluation/embedding_metrics.py

```diff
@@ -0,0 +1,142 @@
+from skills_ml.ontologies.clustering import Clustering
+
+from collections import defaultdict
+from scipy.spatial import distance
+from gensim.models import KeyedVectors
+from typing import Callable, Dict
+import numpy as np
+import tempfile
+import os
+
+
+def embedding_evaluate(vectorization_pipeline, metric_objects):
+    result = defaultdict(dict)
+    for metric in metric_objects:
+        result[metric.name] =  metric.eval(vectorization_pipeline)
+    return result
+
+
+class BaseEmbeddingMetric(object):
+    def __init__(self):
+        self.eval_result = None
+
+    def eval(self):
+        return NotImplementedError
+
+
+class CategorizationMetric(BaseEmbeddingMetric):
```

Comment:
Docstrings for all of these! I honestly don't know what most of them are, so even I'm going to need the docstrings to understand them